src/core.c

```diff
@@ -277,10 +278,15 @@ ROX_INTERNAL RoxStateCode rox_core_setup(
             if (!api_key || !api_key[0]) {
                 ROX_ERROR("Invalid rollout apikey - must be specified");
                 return RoxErrorEmptyApiKey;
-            } else if (!str_matches(api_key, mongoIdPattern, PCRE2_CASELESS) && !str_matches(api_key, mongoIdPattern, PCRE2_CASELESS)) {
+            } else if (!str_matches(api_key, mongoIdPattern, PCRE2_CASELESS) && !str_matches(api_key, uuidIdPattern, PCRE2_CASELESS)) {
                 ROX_ERROR("Illegal rollout apikey");
                 return RoxErrorInvalidApiKey;
             }
+            // if this is a platform key, disable signature verification
+            if (str_matches(api_key, uuidIdPattern, PCRE2_CASELESS)) {
+                rox_options_set_disable_signature_verification(rox_options, true);
+                disableSignature = true;
+            }
```

Comment:
the code should go to Rollout if it's mongoDB id, or to PLATFORM if it's uuid
couldn't see where this is happening

Reply:
Digging more into how this is implemented (uses environment variables), I think this needs to be revisited to be more dynamic period.